frontend/device/generic/device.lua

```diff
@@ -1061,4 +1063,12 @@ function Device:_afterNotCharging()
     UIManager:broadcastEvent(Event:new("NotCharging"))
 end
 
+function Device:storeTimeOfLastPaint()
+    self.lastPaintTime = UIManager:getElapsedTimeSinceBoot()
```

Comment:
The simple CLOCK_MONOTONIC get should be enough, you're already special-casing the after-suspend case, so we don't need suspend tracking in there, too ;).